src/server/api/auth.ts

```diff
@@ -84,6 +85,69 @@ authRouter.post("/register", async (req, res, next) => {
     }
 })
 
+// POST /api/auth/login
+authRouter.post("/login", async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+
+    const user = await prisma.user.findUniqueOrThrow({
+        where: {
+          email: email,
+        },
+      });
+  
+
+    // if (user) {
+    bcrypt.compare(
+      password,
+      user.password,
+      async function (err: Error | undefined, result: boolean) {
+        if (err) {
+          next(err)
+        } else if (result) {
+          // JSON Web Token returned to client
+          // TODO This solves an error, investigate other way to solve
+          const token = ACCESS_TOKEN_SECRET ?
+          jwt.sign({
+            username: user.username,
+            id: user.id,
+          },
+            ACCESS_TOKEN_SECRET) : 
+            next({
+              name: "InvalidAccessTokenSecret", 
+              message: "Access token secret is undefined"
+            });
+
+          res.send({
+            token,
+            user: excludePassword(user),
+          });
+        } else {
+          next({
+            name: "IncorrectPassword",
+            message: "The password you entered is incorrect",
+          });
+        }
+      }
+    )
+  // };
+  } catch (e) {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+      // The .code property can be accessed in a type-safe manner
+      if (e.code === 'P2001') {
+        console.log(e.message)
+      }
+      res.status(401)
+      .send({
```

Comment:
Changing `send()` to `next()` will ensure that we don't get that error of the headers already being sent. Next will pass it on, with the status we set, to the Express error handler.

```suggestion
      res.status(401);
      next({
```